site/migration/01-migration.md

```diff
@@ -0,0 +1,257 @@
+   sections in `v1alpha2` version of Kptfile.
+3. `dependencies` section is removed from Kptfile.
+4. Starlark functions section is removed from Kptfile.
+5. `Setters` and `substitutions` are converted to new and simple setter patterns. The
+   setter comments in the resources are updated to follow new comment syntax.
+6. The `apply-setters` function and all the setters are added to the mutators section
+   in the pipeline.
+7. Function annotation from function configs will be removed and corresponding function
+   definitions will be declared in `pipeline` section of Kptfile. Reference to function config
+   is added via [configPath] option.
+
+Note: This function modifies only the local package files and doesn’t make any
+changes to the resources in the live cluster.
+
+### Manual portion of migration
+
+1. All the functions are treated as `mutators` by the `gcr.io/kpt-fn/fix` function and
+   are added to the `mutators` section in the pipeline. Users must manually go through
+   the functions and move the validator functions to the `validators` section in the
+   pipeline section of `v1alpha2` Kptfile.
+   1. The order of functions also must be re-arranged manually by users if needed.
+   2. Also, note that the [function config] is used to configure the function
+      and is not included in the input list of resources to function.
+2. [OpenAPI validations] and required setters feature offered by `v0.39.0` setters is no
+   longer offered in `v1.0` version of kpt. Users must write their own validation functions
+   to achieve the functionality.
+   `Tip:` Adding a [starlark function] would be an easier alternative to achieve the
+   equivalent validation functionality.
+3. If you have used Starlark runtime in `v0.39.0`, please checkout the new and improved
+   [Starlark functions] and declare it in the pipeline.
+4. [Auto-setters] feature is deprecated in `v1.0` version of kpt. Since the setters are
+   migrated to a new and simple declarative version, package consumers can easily
+   declare all the setter values and render them all at once.
+5. The `gcr.io/kpt-fn/fix` function doesn’t migrate the `inventory object` to `ResourceGroup`.
+   If you are using the inventory object to manage live resources, please refer to
+   `live migrate` command docs to perform [live migration] separately.
+
+Finally, test your migrated kpt package end-to-end and make sure that the functionality
+is as expected. `gcr.io/kpt-fn/fix` is a helper for migration and doesn't guarantee
+functional parity.
+
+## Timeline
+
+1. `Right now:` You can install and try the pre-release version of kpt `v1.0` binary by following the documentation.
+2. `June 1, 2021:` `gcr.io/kpt-fn/fix` function will be released in kpt-functions-catalog.
+   You can start migrating your existing kpt packages using the function.
+3. `July 1, 2021:` kpt `v1.0` binary will be released. The existing kpt packages are not
```

Comment:
I would reword this: The package format schema will be granduated to v1.
The semver of the CLI binary can be orthogonal to this. In other words, we can guarantee backwards compatibility of package format earlier than 1.0 release of the CLI.

Reply:
I feel the wording you suggested is complex. I am having trouble understanding it.

Reply:
There are two separate independent releases: The package format (ie.g. Kptfile and other CRD) can be graduated to apiVersion v1 *before* we release CLI version 1.0.

Reply:
So we have different timeline for both ? Can you make please a separate PR using edit icon after this PR is merged?